internal/config/discoverer_test.go

```diff
@@ -134,47 +179,93 @@ func TestDiscovererClient_Bind(t *testing.T) {
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       fields: fields {
-		           Duration: "",
-		           Client: GRPCClient{},
-		           AgentClientOptions: GRPCClient{},
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
+		func() test {
+			return test{
+				name: "return DiscovererClient when the bind successes",
+				fields: fields{
+					Duration: "10ms",
+				},
+				want: want{
+					want: &DiscovererClient{
+						Duration:           "10ms",
+						Client:             newGRPCClientConfig(),
+						AgentClientOptions: newGRPCClientConfig(),
+					},
+				},
+			}
+		}(),
+		func() test {
+			c := new(GRPCClient)
+			ac := new(GRPCClient)
+
+			return test{
+				name: "return DiscovererClient when the bind successes and Client and AgentClientOptions is not nil",
+				fields: fields{
+					Duration:           "10ms",
+					Client:             c,
+					AgentClientOptions: ac,
+				},
+				want: want{
+					want: &DiscovererClient{
+						Duration: "10ms",
+						Client: &GRPCClient{
+							ConnectionPool: new(ConnectionPool),
+							DialOption: &DialOption{
+								Insecure: true,
+							},
+							TLS: new(TLS),
+						},
+						AgentClientOptions: &GRPCClient{
+							ConnectionPool: new(ConnectionPool),
+							DialOption: &DialOption{
+								Insecure: true,
+							},
+							TLS: new(TLS),
+						},
+					},
+				},
+			}
+		}(),
+		func() test {
+			key := "DURATION"
+			val := "10ms"
 
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           fields: fields {
-		           Duration: "",
-		           Client: GRPCClient{},
-		           AgentClientOptions: GRPCClient{},
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+			return test{
+				name: "return DiscovererClient when the bind successes and the data is loaded from the environment variable",
+				fields: fields{
+					Duration: "_" + key + "_",
+				},
+				beforeFunc: func(t *testing.T) {
+					t.Helper()
+					if err := os.Setenv(key, val); err != nil {
+						t.Fatal(err)
+					}
+				},
+				afterFunc: func(t *testing.T) {
+					t.Helper()
+					if err := os.Unsetenv(key); err != nil {
+						t.Fatal(err)
+					}
+				},
+				want: want{
+					want: &DiscovererClient{
+						Duration:           "10ms",
+						Client:             newGRPCClientConfig(),
+						AgentClientOptions: newGRPCClientConfig(),
```

Comment:
Please ussing 
```golang
&GRPCClient{
    DialOption: &DialOption{
        Inserure: true,
    }
}
```
instead of those implement

Reply:
thank you.
I fixed.